lib/rust/ensogl/component/grid-view/src/lib.rs

```diff
@@ -0,0 +1,508 @@
+        /// event for given row and column. After that the entry will be visible.
+        model_for_entry(Row, Col, EntryModel),
+        /// Set the entries size. All entries have the same size.
+        set_entries_size(Vector2),
+        /// Set the entries parameters.
+        set_entries_params(EntryParams),
+        /// Set the layer for any texts rendered by entries. The layer will be passed to entries'
+        /// constructors. **Performance note**: This will re-instantiate all entries.
+        set_text_layer(Option<WeakLayer>),
+    }
+
+    Output {
+        row_count(Row),
+        column_count(Col),
+        viewport(Viewport),
+        entries_size(Vector2),
+        entries_params(EntryParams),
+        content_size(Vector2),
+        /// Event emitted when the Grid View needs model for an uncovered entry.
+        model_for_entry_needed(Row, Col),
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+// === EntryCreationCtx ===
+
+#[derive(CloneRef, Debug, Derivative)]
+#[derivative(Clone(bound = ""))]
+struct EntryCreationCtx<P> {
+    app:              Application,
+    network:          frp::WeakNetwork,
+    set_entry_size:   frp::Stream<Vector2>,
+    set_entry_params: frp::Stream<P>,
+}
+
+impl<P: frp::node::Data> EntryCreationCtx<P> {
+    fn create_entry<E: Entry<Params = P>>(&self, text_layer: &Option<Layer>) -> E {
+        let entry = E::new(&self.app, text_layer);
+        match self.network.upgrade() {
+            Some(network) => {
+                let entry_frp = entry.frp();
+                let entry_network = entry_frp.network();
+                frp::new_bridge_network! { [network, entry_network] grid_view_entry_bridge
+                    init <- source_();
+                    entry_frp.set_size <+ all(init, self.set_entry_size)._1();
+                    entry_frp.set_params <+ all(init, self.set_entry_params)._1();
+                }
+                init.emit(());
+            }
+            None => {
+                tracing::warn!("Tried to connect entry FRP when the GridView network is dropped");
+            }
+        }
+        entry
+    }
+}
+
+fn set_entry_position<E: display::Object>(entry: &E, row: Row, col: Col, entry_size: Vector2) {
+    let x = (col as f32 + 0.5) * entry_size.x;
+    let y = (row as f32 + 0.5) * -entry_size.y;
+    entry.set_position_xy(Vector2(x, y));
+}
+
+
+// === Properties ===
+
+#[derive(Copy, Clone, Debug, Default)]
+struct Properties {
+    row_count:    usize,
+    col_count:    usize,
+    viewport:     Viewport,
+    entries_size: Vector2,
+}
+
+
+// === Model ===
+
+/// The Model of Select Component.
+#[derive(Clone, Debug)]
+pub struct Model<E, P> {
+    display_object:     display::object::Instance,
+    visible_entries:    RefCell<HashMap<(Row, Col), E>>,
+    free_entries:       RefCell<Vec<E>>,
+    entry_creation_ctx: EntryCreationCtx<P>,
+}
```

Comment:
You are using `<E, P>` here instead of `<E::Entry>`, right? 
1. Thank you so much for caring about it.
2. I think that this might be a corner case where we could actually allow `<E::Entry>`. Let's have a short call about it tmrw :)

Reply:
After a call, we decided to keep `<Entry, EntryParams>` instead of `<E: Entry>`.